src/class.jl

```diff
@@ -0,0 +1,72 @@
+"""
+    Class
+
+A type used for controlling which classes of `TimeZone` are considered valid. Instances of
+`Class` can be combined using bitwise operators to generate a mask which allows multiple
+classes to be considered valid at once.
+
+Currently supported masks are:
+
+- `Class.FIXED`: Class indicating the time zone name is parsable as a fixed UTC offset.
+- `Class.STANDARD`: The time zone name is included in the primary IANA tz source files.
+- `Class.LEGACY`: The time zone name is included in the deprecated IANA tz source files.
+- `Class.NONE`: Mask that will match nothing.
+- `Class.DEFAULT`: Default mask used by functions: `Class.FIXED | Class.STANDARD`
+- `Class.ALL`: Mask allowing all supported classes.
+"""
+struct Class
+    val::UInt8
+end
+
+function Base.getproperty(::Type{Class}, field::Symbol)
```

Comment:
This part seems a little awkward to me. I'd probably just use constants that aren't exported.

```
const NONE = Class(0x00)
...
const DEFAULT = FIXED | STANDARD | LEGACY
...
```

Reply:
Alternatively, maybe you could do `Class[:STANDARD]`, `Class("STANDARD")` or `class"STANDARD"` just to make it clear that `Class` isn't a module when you read `Class.STANDARD`?

Reply:
Alternatively, we can just make it a module:
```julia
baremodule Class
const NONE = 0x00
const FIXED = 0x01
...
end
```
Then you get `Class.FIXED` syntax for free and don't need to define any special bitwise operations for `Class` objects since the actual values you're using already have those defined.

Reply:
True. You'd be able to recover some of that with
```julia
baremodule Class
@enum TZClass begin
    const NONE = 0x00
    ...
end
end
```
but that's getting into unnecessary complication territory.

Reply:
I'm fine with this use of `getproperty` but since there is some controversy I'm going to switch `Class.STANDARD` to `Class(:STANDARD)`. 

I'm not a fan of `Class[:STANDARD]` as that makes me feel like this is an instance I can pull values out of like an array or a dataframe. `Class("STANDARD")` is ok but I already have a constructor which uses the time zone name as a string.